Core/src/org/sleuthkit/autopsy/mainui/datamodel/DataArtifactDAO.java

```diff
@@ -154,6 +174,94 @@ public void dropDataArtifactCache() {
+                        }
+                    } catch (TskCoreException ex) {
+                        logger.log(Level.WARNING, "Unable to fetch artifact category for artifact with id: " + art.getId(), ex);
+                    }
+                    return null;
+                })
+                .filter(pr -> pr != null)
+                .collect(Collectors.groupingBy(pr -> pr.getKey(), Collectors.mapping(pr -> pr.getValue(), Collectors.toSet())));
+
+        // don't do anything else if no relevant events
+        if (artifactTypeDataSourceMap.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        // invalidate cache entries that are affected by events
+        ConcurrentMap<SearchParams<BlackboardArtifactSearchParam>, DataArtifactTableSearchResultsDTO> concurrentMap = this.dataArtifactCache.asMap();
+        concurrentMap.forEach((k, v) -> {
+            Set<Long> dsIds = artifactTypeDataSourceMap.get(k.getParamData().getArtifactType());
+            if (dsIds != null) {
+                Long searchDsId = k.getParamData().getDataSourceId();
+                if (searchDsId == null || dsIds.contains(searchDsId)) {
+                    concurrentMap.remove(k);
+                }
+            }
+        });
+
+        // gather dao events based on artifacts
+        List<DataArtifactEvent> dataArtifactEvents = new ArrayList<>();
+        for (Entry<BlackboardArtifact.Type, Set<Long>> entry : artifactTypeDataSourceMap.entrySet()) {
+            BlackboardArtifact.Type artType = entry.getKey();
+            for (Long dsObjId : entry.getValue()) {
+                DataArtifactEvent newEvt = new DataArtifactEvent(artType, dsObjId);
+                dataArtifactEvents.add(newEvt);
+            }
+        }
+
+        List<TreeEvent> newTreeEvents = this.treeCounts.enqueueAll(dataArtifactEvents).stream()
+                .map(daoEvt -> new TreeEvent(getTreeItem(daoEvt.getArtifactType(), daoEvt.getDataSourceId(), TreeDisplayCount.INDETERMINATE), false))
+                .collect(Collectors.toList());
+
+        return Stream.of(dataArtifactEvents, newTreeEvents)
+                .flatMap((lst) -> lst.stream())
+                .collect(Collectors.toList());
+    }
+
+    private TreeItemDTO<DataArtifactSearchParam> getTreeItem(BlackboardArtifact.Type artifactType, Long dataSourceId, TreeDisplayCount displayCount) {
+        return new TreeResultsDTO.TreeItemDTO<>(
+                BlackboardArtifact.Category.DATA_ARTIFACT.name(),
+                new DataArtifactSearchParam(artifactType, dataSourceId),
+                artifactType.getTypeID(),
+                artifactType.getDisplayName(),
+                displayCount);
+    }
+
+    @Override
+    Collection<? extends DAOEvent> handleIngestComplete() {
+        return this.treeCounts.flushEvents().stream()
+                .map(daoEvt -> new TreeEvent(getTreeItem(daoEvt.getArtifactType(), daoEvt.getDataSourceId(), TreeDisplayCount.INDETERMINATE), true))
```

Comment:
Should this be INDETERMINITE?

Reply:
@bcarrier , this event notifies the UI that it should request an exact count from the DAO's for the given search params (a.k.a. the artifact type and data source).   The INDETERMINATE count isn't used in the UI for this event.  Is this okay as is?

Reply:
If it is going to be ignored, then I'd prefer a more explicit value to avoid confusion.  Something like making a TreeDisplayCount.UNSPECIFIED.